plugin-dotnet-agent/src/main/kotlin/jetbrains/buildServer/custom/CustomCommandWorkflowComposer.kt

```diff
@@ -42,19 +44,36 @@ class CustomCommandWorkflowComposer(
             return Workflow()
         }
 
+        return Workflow(createCommandLines(context))
+    }
+
+    private fun createCommandLines(context: WorkflowContext) = sequence {
         val workingDirectory = Path(_pathsService.getPath(PathType.WorkingDirectory).path)
-        val args = parameters(DotnetConstants.PARAM_ARGUMENTS)?.trim()?.let {
-            _argumentsService.split(it).map { CommandLineArgument(it, CommandLineArgumentType.Custom) }.toList()
-        } ?: emptyList<CommandLineArgument>()
+        val args = parameters(DotnetConstants.PARAM_ARGUMENTS)?.trim()?.let { argString ->
+            _argumentsService.split(argString).map {
+                CommandLineArgument(it, CommandLineArgumentType.Custom)
+            }.toList()
+        } ?: emptyList()
 
         val targets = _targetService.targets.map { it.target }.toMutableList()
         if (targets.isEmpty()) {
             // to run dotnet tools
             targets.add(Path(""))
         }
 
-        return Workflow(targets.asSequence().map { CommandLine(null, target, it, workingDirectory, args) })
+        context.toExitCodes()
+            .subscribe {
+                if (it != 0 && _buildOptions.failBuildOnExitCode) {
+                    context.abort(BuildFinishedStatus.FINISHED_FAILED)
```

Comment:
This works fine but it will be better to report a reason of failure i.e. to report a build problem similar [this one](https://github.com/JetBrains/teamcity-dotnet-plugin/blob/7fc656c829308e9b99dfff67bd964d9418c4fa63/plugin-dotnet-agent/src/main/kotlin/jetbrains/buildServer/dotnet/DotnetWorkflowAnalyzerImpl.kt#L27) in addition to boring and setting failed status. It would be useful because right now you can see build failure reason as a default build problem "Unknown build problem reported on agent". That looks not really informative.

Reply:
Just for comparison

### 1. How it looks now
<img width="567" alt="Screenshot 2022-10-28 at 13 28 00" src="https://user-images.githubusercontent.com/4894495/198577073-6df44004-0836-47c3-85f3-362abd8e87ac.png">

### 2. How it can looks
<img width="567" alt="Screenshot 2022-10-28 at 13 28 15" src="https://user-images.githubusercontent.com/4894495/198577106-bdd3fd08-2b62-4bcb-ae09-eb99ef4f3887.png">

### 3. On a build configuration Overview page
<img width="471" alt="Screenshot 2022-10-28 at 13 28 35" src="https://user-images.githubusercontent.com/4894495/198577169-9cb01e05-017c-43c9-ac07-4e4a73d1b87b.png">

Reply:
done